cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -179,6 +328,70 @@ size_type inplace_bitmask_binop(Binop op,
   return d_counter.value(stream);
 }
 
+/**
+ * @brief Performs a segmented bitwise operation `op` across multiple bitmasks and writes the
+ * results in-place to destination masks.
+ *
+ * This function performs bitwise operations on segments of bitmasks defined by segment_offsets,
+ * writing the results directly to the specified destination masks.
```

Comment:
Please feel free to improve this. I feel like the meaninng changes when we say segmented op on bitmasks vs op on segments of bitmasks. The latter one seems more accurate to me.

```suggestion
 * @brief Performs a bitwise operation `op` across multiple segments of input bitmasks and writes the
 * results from each segment in-place to destination masks.
 *
 * This function performs bitwise operations on segments of bitmasks defined by segment_offsets,
 * writing the results directly to the specified destination masks.
```